### PR TITLE
Add environment variable expire web session

### LINKED ISSUE #4107: Shorten expiration of session on WebUI
**What would you like to be added**:
Shorten the expiration of session on WebUI.

**Why is this needed**:
Long expiration of session induces security risks.

### PR DESCRIPTION
**Pull request by kevin55156**
**What this PR does / why we need it**:
Enabling developers change session ttl with config file

**Which issue(s) this PR fixes**:

Fixes #4107 

**Does this PR introduce a user-facing change?**:

```release-note
Add field "sessionTtl" to sharedSSOConfig
```

### CODE REVIEW

File: docs/content/en/docs-dev/user-guide/managing-controlplane/configuration-reference.md [hunk 1]
```diff
@@ -147,6 +147,7 @@ Must be one of the following objects:
 |-|-|-|-|
 | name | string | The unique name of the configuration. | Yes |
 | provider | string | The SSO service provider. Can be one of the following values<br>`GITHUB`, `GOOGLE`... | Yes |
+| sessionTtl | int | The time to live of session for sso login | Yes |
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-controlplane/configuration-reference.md:150]:
```suggestion
| sessionTtl | int | The time to live of session for SSO login. Unit is `minute`. | Yes |
```

File: quickstart/control-plane-values.yaml [hunk 2]
```diff
@@ -6,6 +6,8 @@ config:
     apiVersion: "pipecd.dev/v1beta1"
     kind: ControlPlane
     spec:
+      stateKey: gnvraeiegrojblskgnlhej
+      address: http://127.0.0.1:8080
```
Comment (khanhtc1202) [inline quickstart/control-plane-values.yaml:10]: Could you remove this 👀

File: pkg/model/project.proto [hunk 3]
```diff
@@ -97,6 +97,8 @@ message ProjectSSOConfig {
     }
 
     Provider provider = 1 [(validate.rules).enum.defined_only = true];
+    // session ttl for users (munites)
+    int64 session_ttl = 2 [(validate.rules).int64.gt = 0];
```
Comment (khanhtc1202) [inline pkg/model/project.proto:101]: With the rule set like this, I guess project object which is stored in datastore currently without this value will cause error, right?

Comment (kevin55156) [reply pkg/model/project.proto:101]: When server take configuration file in, sever does not use getter generated by grpc, so validation does not works.
If the field of sessionTtl does not exist in config file, the zero is set at that time.

File: docs/content/en/docs-dev/user-guide/managing-controlplane/configuration-reference.md [hunk 4]
```diff
@@ -147,6 +147,7 @@ Must be one of the following objects:
 |-|-|-|-|
 | name | string | The unique name of the configuration. | Yes |
 | provider | string | The SSO service provider. Can be one of the following values<br>`GITHUB`, `GOOGLE`... | Yes |
+| sessionTtl | int | The time to live of session for SSO login. Unit is `minute`. | Yes |
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-controlplane/configuration-reference.md:150]: Do we have any reason to mark this field as required? We have a default value for this setting anyway.

File: docs/content/en/docs-dev/user-guide/managing-controlplane/configuration-reference.md [hunk 5]
```diff
@@ -147,6 +147,7 @@ Must be one of the following objects:
 |-|-|-|-|
 | name | string | The unique name of the configuration. | Yes |
 | provider | string | The SSO service provider. Can be one of the following values<br>`GITHUB`, `GOOGLE`... | Yes |
+| sessionTtl | int | The time to live of session for SSO login. Unit is `minute`. Default is 7 hours. | No |
```
Comment (khanhtc1202) [inline docs/content/en/docs-dev/user-guide/managing-controlplane/configuration-reference.md:150]: I think the current default TTL is 7 days
ref: https://github.com/pipe-cd/pipecd/blob/master/pkg/app/server/httpapi/auth_handler.go#L51

Comment (khanhtc1202) [reply docs/content/en/docs-dev/user-guide/managing-controlplane/configuration-reference.md:150]: And looks like the current default time is in units of hours, so could you recheck around that part?

Comment (kevin55156) [reply docs/content/en/docs-dev/user-guide/managing-controlplane/configuration-reference.md]: Thank you, I have changed "minutes" to "hours"

**Review by khanhtc1202 — APPROVED** [review]
Nice work, thanks 🙌

**Pull request merged**